wlauto/workloads/stress_ng/__init__.py

```diff
@@ -0,0 +1,96 @@
+
+
+class StressNg(Workload):
+
+    name = 'stress_ng'
+    description = """
+    stress-ng will stress test a computer system in various selectable ways. It
+    was designed to exercise various physical subsystems of a computer as well
+    as the various operating system kernel interfaces.
+
+    stress-ng can also measure test throughput rates; this can be useful to
+    observe performance changes across different operating system releases or
+    types of hardware. However, it has never been intended to be used as a
+    precise benchmark test suite, so do NOT use it in this manner.
+
+    The official website for stress-ng is at:
+        http://kernel.ubuntu.com/~cking/stress-ng/
+
+    Source code are available from:
+        http://kernel.ubuntu.com/git/cking/stress-ng.git/
+    """
+
+    parameters = [
+        Parameter('stressor', kind=str, default='cpu',
+                  allowed_values=['cpu', 'io', 'fork', 'switch', 'vm', 'pipe',
+                                  'yield', 'hdd', 'cache', 'sock', 'fallocate',
+                                  'flock', 'affinity', 'timer', 'dentry',
+                                  'urandom', 'sem', 'open', 'sigq', 'poll'],
+                  description='Stress test case name. The cases listed in '
+                              'allowed values come from the stable release '
+                              'version 0.01.32. The binary included here '
+                              'compiled from dev version 0.06.01. Refer to '
+                              'man page for the definition of each stressor.'),
+        Parameter('threads', kind=int, default=0,
+                  description='The number of workers to run. Specifying a '
+                              'negative or zero value will select the number '
+                              'of online processors.'),
+        Parameter('duration', kind=int, default=60,
+                  description='Timeout for test execution in seconds')
+    ]
+
+    def setup(self, context):
+        host_binary = context.resolver.get(Executable(self, self.device.abi,
+                                                      'stress-ng'))
+        self.binary = self.device.install_if_needed(host_binary)
+        self.log = os.path.join(self.device.working_directory,
+                                'stress_ng_output.txt')
+        self.results = os.path.join(self.device.working_directory,
+                                    'stress_ng_results.yaml')
+        self.command = ('{} --{} {} --timeout {}s --log-file {} --yaml {} '
+                        '--metrics-brief --verbose'
+                        .format(self.binary, self.stressor, self.threads,
+                                self.duration, self.log, self.results))
+        self.timeout = self.duration + 10
+
+    def run(self, context):
+        self.output = self.device.execute(self.command, timeout=self.timeout,
+                                          as_root=True)
```

Comment:
Since you're running as root, you should add an `initialize()` method and in it check that the device is rooted (`self.device.is_rooted`), raising `WorkloadError` if it isn't.